CMakeLists.txt

```diff
@@ -239,6 +250,81 @@ endif()
 
 include(cmake/optional.cmake)
 
+set(core_cxx_flags)
+# Don't allow extended (non-ASCII) symbols in identifiers. This is easier for code review.
+try_append_cxx_flags(core_cxx_flags "-fno-extended-identifiers")
+if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
```

Comment:
This is still the wrong thing todo. It doesn't matter what is implied by the comment.

We don't want to perform checks based on some compiler string, we want generic checks, that can be run against any compiler, and overall less branching in build code.

For example, if you gate things here based on "GNU" (GCC), and at some point someone forks GCC, and renames it, any users of that new compiler now get a worse, or broken experience, because things are unnecessarily hardcoded to check for compiler names. 

Or, maybe Clang will add this same flag, and now our check is also deficient, or we aren't disabling it when we should be.

Generic checks will not fail in this way, and are much more forward-compatible.

Reply:
I will remove the `CMAKE_CXX_COMPILER_ID STREQUAL "GNU"` check in the next push.

However, I don't quite agree with all of your points.

> We don't want to perform checks based on some compiler string, we want generic checks, that can be run against any compiler, and overall less branching in build code.

Agree. But in this particular case, the generic check should test a compiler whether it has a bug in the `-fstack-reuse` implementation. That is not the case here.

> ... at some point someone forks GCC, and renames it, any users of that new compiler now get a worse, or broken experience

That is a very extreme corner case and I doubt whether it should be a matter of concerns.

> Or, maybe Clang will add this same flag, and now our check is also deficient, or we aren't disabling it when we should be.

Why do we assume that Clang will have bugs in the `-fstack-reuse` implementation?

> Generic checks will not fail in this way, and are much more forward-compatible.

True. But passing test "a compiler has `-fstack-reuse=none`" does not logically mean that "a compiler has a bug".

---

Anyway, I will remove the `CMAKE_CXX_COMPILER_ID STREQUAL "GNU"` check in the next push.

Reply:
The same reasoning I've provided holds for basically all other checks we perform. i.e, we don't test for specific warning flags depending on if we know they exist in the compiler, we test generically, with the assumption that if it currently exist in one compiler, it may exist in another in the near future, and scoping to check to a single compiler, provides no benefit.

This is also the exact same issue I pointed out below in regards to [Clang & ld, and using `--exclude-libs,All`](https://github.com/hebasto/bitcoin/pull/32#discussion_r1363581652). Scoping that check to "GNU", doesn't provide any benefit, and was just pointlessly restricting functionality for anyone using Clang.

Reply:
> I will remove the `CMAKE_CXX_COMPILER_ID STREQUAL "GNU"` check in the next push.

Done.